src/components/SortDropdown.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+
+const useStyles = makeStyles((theme) => ({
+  hiddenStyle: {
+    display: 'none',
+  },
+  selectedOptionTextStyle: {
+    fontWeight: '500',
+    fontFamily: 'Work Sans',
+    fontSize: '14px',
+    lineHeight: '18px',
+    margin: 'auto 0',
+  },
+  optionTextStyle: {
+    fontWeight: '500',
+    fontFamily: 'Work Sans',
+    fontSize: '14px',
+    lineHeight: '18px',
+    margin: 'auto 0 auto 10px',
+  },
+  dropdown: {
+    display: 'flex',
+    cursor: 'pointer',
+    backgroundColor: theme.palette.background.default,
+    boxSizing: 'border-box',
+    borderRadius: '4px 4px 0px 0px',
+    minHeight: '43px',
+    minWidth: '177px',
+    border: '1px solid #BCBCBC',
+  },
+  selectedOptionGridStyle: {
+    display: 'inline-flex',
+    marginLeft: '10px',
+  },
+  dropdownIconStyle: {
+    marginLeft: '10px',
+    width: '32px',
+    height: '29px',
+  },
+  iconGridStyle: {
+    margin: 'auto',
+    paddingTop: '6px',
+  },
+  optionContainerStyle: {
+    display: 'flex',
+    minHeight: '43px',
+    minWidth: '177px',
+    backgroundColor: theme.palette.background.default,
+    border: '0.5px solid #BCBCBC',
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.palette.secondary.light,
+    },
+  },
+  optionsGridStyle: {
+    position: 'absolute',
+    borderRadius: '0px 0px 4px 4px',
+    left: '0',
+    right: '0',
+  },
+  dropdownContainer: {
+    position: 'relative',
+    display: 'inline-block',
+  },
+}));
+
+export const SortDropdown = ({
+  inputSortMethodList,
+  defaultSortMethod,
+  setSortMethod,
+}) => {
+  const classes = useStyles();
+  const [isDropdownExpand, setIsDropdownExpand] = useState(false);
+  const [selectedSortMethod, setSelectedSortMethod] =
+    useState(defaultSortMethod);
+  const sortMethodMap = {
+    'best match': 'Best Match',
+    'updated': 'Last Updated',
+    'stars': 'Stargazer Count',
+  };
+
+  const handleChange = (event) => {
+    const sortMethodVal = event.target.dataset.value;
+    setSelectedSortMethod(sortMethodVal);
+    setSortMethod(sortMethodVal);
+    handleDropdownOpen();
+  };
+  const handleDropdownOpen = () => {
+    setIsDropdownExpand(!isDropdownExpand);
+  };
```

Comment:
Since handleChange is calling handleDropdownOpen every time handleChange is run, you can move the line of code from handleDropdownOpen into handleChange. In addition, you can even change the name of the state variable and function. Maybe like toggleDropdown? It seems like this function is trying to open and close the dropdown so just a suggestion.
```suggestion
  const handleChange = (event) => {
    const sortMethodVal = event.target.dataset.value;
    setSelectedSortMethod(sortMethodVal);
    setSortMethod(sortMethodVal);
    toggleDropdown(!isDropdownExpand);
  };
```